src/OpeningHoursSpecificationParser.php

```diff
@@ -0,0 +1,176 @@
+            'Tuesday', 'https://schema.org/Tuesday' => 'tuesday',
+            'Wednesday', 'https://schema.org/Wednesday' => 'wednesday',
+            'Thursday', 'https://schema.org/Thursday' => 'thursday',
+            'Friday', 'https://schema.org/Friday' => 'friday',
+            'Saturday', 'https://schema.org/Saturday' => 'saturday',
+            'Sunday', 'https://schema.org/Sunday' => 'sunday',
+            default => throw new InvalidOpeningHoursSpecification('Invalid https://schema.org Day specification'),
+        };
+    }
+
+    private function addDayOfWeekHours(
+        string $dayOfWeek,
+        ?string $opens,
+        ?string $closes
+    ): void {
+        $dayOfWeek = self::schemaOrgDayToString($dayOfWeek);
+
+        $hours = $this->formatHours($opens, $closes);
+        if (null === $hours) {
+            return;
+        }
+        $this->openingHours[$dayOfWeek][] = $hours;
+    }
+
+    private function addExceptionsHours(
+        ?string $validFrom,
+        ?string $validThrough,
+        ?string $opens,
+        ?string $closes
+    ): void {
+        if (! is_string($validFrom) || ! is_string($validThrough)) {
+            throw new InvalidOpeningHoursSpecification('Missing validFrom and validThrough dates');
+        }
+
+        if (! preg_match('/^\d{4}-\d{2}-\d{2}$/', $validFrom) || ! preg_match('/^\d{4}-\d{2}-\d{2}$/', $validThrough)) {
+            throw new InvalidOpeningHoursSpecification('Invalid validFrom and validThrough dates');
+        }
+
+        if ($validFrom === $validThrough) {
+            $exceptionKey = $validFrom;
+        } else {
+            $exceptionKey = $validFrom.' to '.$validThrough;
+        }
+
+        if (! isset($this->openingHours['exceptions'])) {
+            $this->openingHours['exceptions'] = [];
+        }
+        if (! isset($this->openingHours['exceptions'][$exceptionKey])) {
+            // Default to close all day
+            $this->openingHours['exceptions'][$exceptionKey] = [];
+        }
+
+        $hours = $this->formatHours($opens, $closes);
+        if (null === $hours) {
+            return;
+        }
+        $this->openingHours['exceptions'][$exceptionKey][] = $hours;
+    }
+
+    private function formatHours(
+        ?string $opens,
+        ?string $closes,
+    ): ?string {
+        if (! is_string($opens) || ! is_string($closes)) {
+            throw new InvalidOpeningHoursSpecification('Missing opens and closes hours');
+        }
+
+        if (
+            ! preg_match('/^\d{2}:\d{2}(:\d{2})?$/', $opens) ||
+            ! preg_match('/^\d{2}:\d{2}(:\d{2})?$/', $closes)
+        ) {
+            throw new InvalidOpeningHoursSpecification('Invalid opens and closes hours');
+        }
+
+        // strip seconds part if present
+        $opens = preg_replace('/^(\d{2}:\d{2})(:\d{2})?$/', '$1', $opens);
+        $closes = preg_replace('/^(\d{2}:\d{2})(:\d{2})?$/', '$1', $closes);
+
+        // Ignore 00:00-00:00 which means closed all day
```

Comment:
This sounds quite unintuitive, where is it specified?

Reply:
Are you talking about `00:00-00:00` or stripping seconds from specification ?

Reply:
1. OpeningHoursSpecification can support seconds: \
![image](https://github.com/spatie/opening-hours/assets/380026/0bf211df-72fe-4fdc-a4ef-e01761c89845)

2. For `00:00-00:00` it's specified on Google documentation : https://developers.google.com/search/docs/appearance/structured-data/local-business?hl=fr#business-hours

Reply:
Ah OK, I was searching in https://schema.org/OpeningHoursSpecification

Now I also understand why you called it Google-flavored in the doc.